File: learning/submitted_projects_CS50x/mario-more/mario.py
from cs50 import get_int
n = get_int("height: ")
while n > 8 or n < 1:
    n = get_int("height: ")
for i in range(n):
    for k in range(n-i-1):
        print(" ",end='')
    for j in range(i+1):
        print("#" ,end='')
    print("  ",end='')
    for j in range(i+1):
        print("#",end='')
    print("")

File: learning/submitted_projects_CS50x/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

typedef void (*FunctionPointer)();
void repeat(int n, FunctionPointer func)
{
    for (int i = 0; i < n; i++)
    {
        func();
    }
}

void p();
void spx();

int main()
{
    int x;

    do
    {

        x = get_int("height: ");
    }
    while (x < 1 || x > 8);

    int i;
    for (i = 0; i < x; i++)
    {

        repeat(x - i - 1, spx);
        repeat(1 + i, p);
        printf("\n");
    }
}

void p()
{
    printf("#");
}

void spx()
{
    printf(" ");
}
